### PR TITLE
Review restaurant

### PR DESCRIPTION
**Pull request by TheNguyen0905**

### CODE REVIEW

**Comment by TheNguyen0905** [comment]
ready

File: app/models/review.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Review < ApplicationRecord
+  belongs_to :user
+  scope :lastest, -> {order created_at: :desc}
```
Comment (mdmahmudurrahman) [inline app/models/review.rb]: `scope :lastest, ->{order created_at: :desc}`

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -33,8 +33,8 @@
         <nav class="h-100 navbar navbar-expand-lg align-items-center">
           <%= link_to t(".brand"), root_path, class: "navbar-brand" %>
           <button class="navbar-toggler" type="button" data-toggle="collapse"
-           data-target="#caviarNav" aria-controls="caviarNav"
-            aria-expanded="false" aria-label="Toggle navigation">
+                  data-target="#caviarNav" aria-controls="caviarNav"
+                  aria-expanded="false" aria-label="Toggle navigation">
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Indentation

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -67,21 +67,17 @@
...
-                  <%= link_to t(".setting"), "#", class: "dropdown-item" %>
-                  <%= link_to t(".review"), "#", class: "dropdown-item",
-                   data: {toggle: "modal", target: "#modal-review"} %>
-                  <%= link_to t(".logout"), logout_path, method: :delete, class: "dropdown-item" %>
-                </div>
-              </li>
+                <li class="nav-item dropdown">
+                  <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown"
+                     role="button" data-toggle="dropdown" aria-haspopup="true"
+                     aria-expanded="false"><%= t ".account" %></a>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: 2 space indentation

File: app/views/reviews/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<!-- ***** Breadcumb Area Start ***** -->
+<div class="breadcumb-area bg-img bg-overlay"
+     style="background-image: url(<%= asset_path('hero-4.jpg') %>)">
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: 2 space indentation

File: app/views/reviews/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        <div class="breadcumb-content">
+          <h2>User Review</h2>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+<!-- ***** Breadcumb Area End ***** -->
+
+<!-- ****** Form Area Start ****** -->
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: Please remove all comments

File: app/views/reviews/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<!-- ***** Breadcumb Area Start ***** -->
+<div class="breadcumb-area bg-img bg-overlay"
+     style="background-image: url(<%= asset_path('hero-4.jpg') %>)">
+  <div class="container h-100">
+    <div class="row h-100 align-items-center">
+      <div class="col-12">
+        <div class="breadcumb-content">
+          <h2>User Review</h2>
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: I18n

File: app/views/reviews/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  </div>
+</div>
+<!-- ***** Breadcumb Area End ***** -->
+
+<!-- ****** Form Area Start ****** -->
+<section class="caviar-reservation-area d-md-flex align-items-center" id="reservation">
+  <div class="reservation-form-area d-flex justify-content-end">
+    <div class="reservation-form">
+      <div class="section-heading">
+        <h2>Leave a review</h2>
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: I18n

File: app/views/reviews/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+          </div>
+        </div>
+      <% end %>
+    </div>
+  </div>
+  <div class="reservation-side-thumb wow fadeInRightBig" data-wow-delay="0.5s">
+    <%= image_tag("hero-5.jpg", alt: "") %>
+  </div>
+</section>
+<!-- ****** Form Area End ****** -->
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: Remove comment

File: app/views/static_pages/home.html.erb [hunk 9]
```diff
@@ -191,38 +191,18 @@
           </div>
           <div class="caviar-testimonials-slides owl-carousel">
             <!-- Single Testimonial Area -->
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Remove

File: app/views/static_pages/home.html.erb [hunk 10]
```diff
@@ -191,38 +191,18 @@
...
-                <div class="tes-name">
-                  <h5>Robert Jonson</h5>
-                  <p>lorem ipsum</p>
+            <% @reviews.each do |review| %>
+              <div class="single-testimonial">
+                <div class="testimonial-thumb-name d-flex align-items-center">
+                  <%= image_tag("3.jpg", alt: "") %>
+                  <div class="tes-name">
+                    <h5><%= review.user.name %></h5>
+                    <p>lorem ipsum</p>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: If it is necessary, use I18n

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 11]
```diff
@@ -1,41 +1,41 @@
...
-  stylers: [{
-    saturation: -100
-      }]
-    }];
+    featureType: "all",
+    elementType: "all",
+    stylers: [{
+        saturation: -100
+            }]
+        }];
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: 1. Please check indentation
2. Use single quote in JS

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 12]
```diff
@@ -1,41 +1,41 @@
...
-    mapTypeIds: [google.maps.MapTypeId.ROADMAP, 'gMap']
-  }
+    zoom: 15,
+    center: latlng,
+    scrollwheel: false,
+    scaleControl: false,
+    disableDefaultUI: true,
+    mapTypeControlOptions: {
+        mapTypeIds: [google.maps.MapTypeId.ROADMAP, 'gMap']
+    }
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: 2 space indentation

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 13]
```diff
@@ -1,41 +1,41 @@
...
 };
 map = new google.maps.Map(document.getElementById("googleMap"), mapOptions);
 var geocoder_map = new google.maps.Geocoder();
 var address = 'London';
 geocoder_map.geocode({
-  'address': address
+    'address': address
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: indentation

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 14]
```diff
@@ -1,41 +1,41 @@
...
+    if (status == google.maps.GeocoderStatus.OK) {
+        map.setCenter(results[0].geometry.location);
+        var marker = new google.maps.Marker({
+            map: map,
+            icon: 'img/core-img/point.png',
+            position: map.getCenter()
+        });
+    } else {
+        alert("Geocode was not successful for the following reason: " + status);
+    }
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: the same here

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 15]
```diff
@@ -1,41 +1,41 @@
...
-  }
+    if (status == google.maps.GeocoderStatus.OK) {
+        map.setCenter(results[0].geometry.location);
+        var marker = new google.maps.Marker({
+            map: map,
+            icon: 'img/core-img/point.png',
+            position: map.getCenter()
+        });
+    } else {
+        alert("Geocode was not successful for the following reason: " + status);
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: 1. I18n
2. Use single quote in JS

File: vendor/assets/javascripts/js/google-map/map-active.js [hunk 16]
```diff
@@ -1,41 +1,41 @@
...
+    }
 });
 var mapType = new google.maps.StyledMapType(stylez, {
-  name: "Grayscale"
+    name: "Grayscale"
```
Comment (mdmahmudurrahman) [inline vendor/assets/javascripts/js/google-map/map-active.js]: The same here

File: app/views/reviews/new.html.erb [hunk 17]
```diff
@@ -0,0 +1,40 @@
+<!-- ***** Breadcumb Area Start ***** -->
```
Comment (mdmahmudurrahman) [inline app/views/reviews/new.html.erb]: Remove comment

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by TheNguyen0905** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 18]
```diff
@@ -33,8 +33,8 @@
         <nav class="h-100 navbar navbar-expand-lg align-items-center">
           <%= link_to t(".brand"), root_path, class: "navbar-brand" %>
           <button class="navbar-toggler" type="button" data-toggle="collapse"
-           data-target="#caviarNav" aria-controls="caviarNav"
-            aria-expanded="false" aria-label="Toggle navigation">
+            data-target="#caviarNav" aria-controls="caviarNav"
+              aria-expanded="false" aria-label="Toggle navigation">
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]:
```ruby
<button class="navbar-toggler" type="button" data-toggle="collapse"
  data-target="#caviarNav" aria-controls="caviarNav"
  aria-expanded="false" aria-label="Toggle navigation">
```

File: app/views/layouts/_header.html.erb [hunk 19]
```diff
@@ -67,21 +67,17 @@
...
-                  <%= link_to t(".setting"), "#", class: "dropdown-item" %>
-                  <%= link_to t(".review"), "#", class: "dropdown-item",
-                   data: {toggle: "modal", target: "#modal-review"} %>
-                  <%= link_to t(".logout"), logout_path, method: :delete, class: "dropdown-item" %>
-                </div>
-              </li>
+                <li class="nav-item dropdown">
+                  <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown"
+                     role="button" data-toggle="dropdown" aria-haspopup="true"
+                       aria-expanded="false"><%= t ".account" %></a>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]:
```ruby
<a class="nav-link dropdown-toggle" href="#" id="navbarDropdown"
  role="button" data-toggle="dropdown" aria-haspopup="true"
  aria-expanded="false"><%= t ".account" %></a>
```

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by TheNguyen0905** [comment]
ready

File: app/controllers/static_pages_controller.rb [hunk 20]
```diff
@@ -1,3 +1,5 @@
 class StaticPagesController < ApplicationController
-  def home; end
+  def home
+    @reviews = Review.lastest.list
```
Comment (huuphuc93) [inline app/controllers/static_pages_controller.rb]: đây là lấy ra thằng cuối cùng à em? @reviews là 1 obj hay 1 collections

File: app/models/user.rb [hunk 21]
```diff
@@ -1,4 +1,5 @@
 class User < ApplicationRecord
+  has_many :reviews
   before_save {email.downcase!}
   validates :name,  presence: true, length: {maximum: Settings.name_max_length}
```
Comment (huuphuc93) [inline app/models/user.rb]: dùng 1 spaces giữ name và presence

**Comment by huuphuc93** [comment]
comment nhỏ

**Comment by huuphuc93** [comment]
commented
comment nhỏ

**Pull request merged**